### PR TITLE
Initial work for CI

### PR DESCRIPTION
**Pull request by anubhavmishra**
PR adds an initial GH Action workflow for CI.

### CODE REVIEW

File: web/yarn.lock [hunk 1]
```diff
@@ -10701,6 +10701,45 @@ __metadata:
   languageName: node
   linkType: hard
 
+"heimdalljs-fs-monitor@npm:^1.1.0":
```
Comment (anubhavmishra) [inline web/yarn.lock:10704]: yarn lock file needs updating for the UI build. @jeffdaley could you please take a look why these changes in the lock file were triggered?

Comment (jeffdaley) [reply web/yarn.lock:10704]: I think it's just an alphabetizing change. Dunno how "heimdall" ever placed after "hermes" though...

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    branches: [main]
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: "read"
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
```
Comment (aldy505) [inline .github/workflows/ci.yml:16]: should probably use v3 as v2 uses node 14, and v3 uses node 16

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+name: ci
+
+on:
+  pull_request:
+    branches: [main]
+  push:
+    branches: [main]
```
Comment (aldy505) [inline .github/workflows/ci.yml:7]: I'd prefer it targets every branch, so everyone gets the benefits of build and test commands. If there are other needs for packaging the source code or perform any other specific command that should be run on PR only, or on main branch only, we can create a separate workflow file for that.

```suggestion
on:
  pull_request:
  push:
```

File: .github/workflows/ci.yml (hunk 2 above)
Comment (anubhavmishra) [reply .github/workflows/ci.yml]: Good catch!

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -21,19 +21,19 @@ jobs:
           node-version: "16"
           cache-dependency-path: "web/yarn.lock"
 
-      - name: yarn set version 3.3.0
-        run: yarn set version 3.3.0
+      - name: yarn set version
+        run: make web/set-yarn-version
```
Comment (anubhavmishra) [inline .github/workflows/ci.yml:25]: Making this a make target so we can track version changes in one file.

File: Makefile [hunk 5]
```diff
@@ -76,3 +81,7 @@ web/run: ## Run web application while proxying backend requests
 web/run: web/install-deps
 	cd web \
 		&& yarn start:with-proxy
+
+web/set-yarn-version: ## Set yarn version
```
Comment (jfreda) [inline Makefile:85]: Nit: could use a `.PHONY` here too.

```suggestion
.PHONY: web/set-yarn-version
web/set-yarn-version: ## Set yarn version
```

**Review by jfreda — APPROVED** [review]

File: Makefile (hunk 5 above)
Comment (anubhavmishra) [reply Makefile:86]: Good catch!
